cluster-autoscaler/core/scaledown/eligibility/eligibility.go

```diff
@@ -135,6 +136,13 @@ func (c *Checker) unremovableReasonAndNodeUtilization(context *context.Autoscali
 		return simulator.NotAutoscaled, nil
 	}
 
+	// In case the node in unready, check if scale down of unready nodes is enabled
+	ready, _, _ := kube_util.GetReadinessState(node)
```

Comment:
nit: `GetNodeReadiness` is a bit more costly than bool flag check, so I'd avoid checking readiness when `ScaleDownUnreadyEnabled` is `true`.

Reply:
@x13n ack, that makes sense. I pushed a fixup on the branch. I put a separate commit for better visibility of the changes, let's squash them all together before merging.